internal/migrate/staging_validator.go

```diff
@@ -552,19 +597,11 @@ func (v *stagingValidatorImpl) resolveLocation(
 	for i := range resolvedLocations {
 		identifier := identifiers[i]
 
-		var resolvedLocation common.Location
-		resovledAddrLocation := common.AddressLocation{
+		resolvedLocation := common.AddressLocation{
 			Address: addressLocation.Address,
 			Name:    identifier.Identifier,
 		}
 
-		// If the contract one of our staged contract updates, use the source location
-		if stagedUpdate, ok := v.stagedContracts[resovledAddrLocation]; ok {
-			resolvedLocation = stagedUpdate.SourceLocation
-		} else {
-			resolvedLocation = resovledAddrLocation
-		}
-
```

Comment:
I think there is no need to separate the staged contracts vs other contracts now, given staged contracts also uses `DeployLocation`, which is an address location.